python/Path.py

```diff
@@ -29,35 +29,35 @@
 temp_invalid_solutions = 0
 
 
-def getPerpLine(lastX, lastY, nextX, nextY):
+def getPerpLine(lastXY, nextXY):
```

Comment:
I like that you made the input args cleaner. Bit of a nitpick, but I find it easier to understand
````
if abs(nextY - lastY) < 0.01:
```

More easily than

```
if abs(nextXY[1] - lastXY[1]) < 0.01:
```

Consider keeping the same args, but then in the first line of the function, make `nextX, nextY = nextXY`, and then using nextX instead of nextXY[0]